src/Components/test/E2ETest/Tests/FormsTest.cs

```diff
@@ -598,6 +598,18 @@ public void CanRemoveAndReAddDataAnnotationsSupport()
             Browser.Equal(new[] { "That name is too long" }, messagesAccessor);
         }
 
+        [Fact]
+        public void InputRangeAttributeOrderDoesNotAffectValue()
```

Comment:
@SteveSandersonMS is it possible to test these sort of changes as vanilla JS tests? I'm not saying we wouldn't have this, but given our perennial struggles with browser testing (and I'm not super sure how much playwrite helps here), is it possible to have some of these additionally covered by JS based unit tests?

Reply:
If you mean "not involving a browser", then I'd say that's not really viable. We're specifically interacting with DOM APIs here so unless there's a real DOM, the best a test could do is something like "assert we called a set of methods on a mock" (which would prove nothing about correct behavior).

We *have* to be able to run browser-based tests. It can't be beyond our abilities as a team/organization! So many other front-end teams can manage this. If our existing infrastructure isn't fit for that purpose, we'll set up new infrastructure.